bpf/bpf_host.c

```diff
@@ -1138,15 +1138,15 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, __u32 __maybe_unused identity,
 		}
 # endif /* defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV6) */
 
-# ifdef ENABLE_WIREGUARD
+# if defined(IS_BPF_HOST) && defined(ENABLE_WIREGUARD)
 		if (!from_host) {
```

Comment:
As you were mentioning, in v1.17 this program will always be compiled with `IS_BPF_HOST`.
Therefore we'd need to check `THIS_INTERFACE_IFINDEX` from within the code (not preprocessor).
```suggestion
		if (!from_host && THIS_INTERFACE_IFINDEX != WG_IFINDEX) {
```